packages/@uppy/core/src/index.js

```diff
@@ -1095,7 +1095,8 @@ class Uppy {
 
       this.setState({ error: errorMsg })
 
-      if (file != null) {
+      // When a file is also given, we store the error on the file object.
+      if (file != null && typeof file === 'object' && typeof file.id === 'string') {
```

Comment:
nit: it's probably safe to skip the `object` test as we already know it's not nullish.
```suggestion
      if (file != null && typeof file.id === 'string') {
```